code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/internal/MessagingConstants.java

```diff
@@ -26,11 +30,27 @@ final class MessagingConstants {
     static final String HTTP_HEADER_IF_NONE_MATCH = "If-None-Match";
     static final String HTTP_HEADER_ETAG = "Etag";
     static final String METADATA_PATH = "pathToFile";
+    static final String ASSET_CACHE_LOCATION = ServiceProvider.getInstance().getDeviceInfoService().getApplicationCacheDir() + File.separator + CACHE_BASE_DIR + File.separator + IMAGES_CACHE_SUBDIRECTORY;
```

Comment:
same as before. i'm a little worried about this chain without any null checks
`ServiceProvider.getInstance().getDeviceInfoService().getApplicationCacheDir()`